orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -357,6 +389,67 @@ def test_input_requires_grad_backward_creates_input_grad(device):
     s.backward()
     assert x.grad is not None
 
+@pytest.mark.parametrize("device", ['cuda', 'cpu'])
+def test_input_requires_grad_backward_creates_input_grad_as_required0(device):
+    N, D_in, H, D_out = 32, 784, 500, 10
+    model = NeuralNetMultiplePositionalArgumentsMultipleOutputs2(D_in, H, D_out).to(device)
+    model = ORTModule(model)
+    x1 = torch.randn(N, D_in, device=device, requires_grad=True)
+    x2 = torch.randn(N, D_in, device=device, requires_grad=True)
+    y1, y2 = model(x1, x2)
+    s = y2.sum()
+    s.backward()
+    assert x1.grad is not None and x2.grad is not None
```

Comment:
a more interesting assert would be 
the param.grad for the y1 branch should be zero.

Reply:
at this stage, y1.grad and y2.grad both are null because the results only contain the gradients of the inputs with require_grad = True. The grad_outputs are not added to results in ortmodule.py. We could add the assertion that unrequired gradient outputs are zero in ortmodule itself.